### PR TITLE
Vlc open in fix

### PR DESCRIPTION
**Pull request by carolanitz**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md) and run `bundle exec fastlane lint` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### Description

This fixes all the open in Issues while still working with the Files app

### CODE REVIEW

File: Sources/VLCPlaybackController.m [hunk 1]
```diff
@@ -142,8 +143,14 @@ - (void)openVideoSubTitlesFromFile:(NSString *)pathToFile
     [_mediaPlayer addPlaybackSlave:[NSURL fileURLWithPath:pathToFile] type:VLCMediaPlaybackSlaveTypeSubtitle enforce:YES];
 }
 
-- (void)playMediaList:(VLCMediaList *)mediaList firstIndex:(NSInteger)index subtitlesFilePath:(NSString *)subsFilePath
+- (void)playMediaList:(VLCMediaList *)mediaList firstIndex:(NSInteger)index subtitlesFilePath:( NSString * _Nullable )subsFilePath
```
Comment (Mikanbu) [inline Sources/VLCPlaybackController.m]: Cosmetic: extra spaces in subtitlesFilePath type~

File: Sources/VLCPlaybackController.m [hunk 2]
```diff
@@ -142,8 +143,14 @@ - (void)openVideoSubTitlesFromFile:(NSString *)pathToFile
...
 {
+    [self playMediaList: mediaList firstIndex: index subtitlesFilePath: subsFilePath completion: nil];
+}
+
+- (void)playMediaList:(VLCMediaList *)mediaList firstIndex:(NSInteger)index subtitlesFilePath:( NSString * _Nullable )subsFilePath completion:(void (^ __nullable)(BOOL success))completion
```
Comment (Mikanbu) [inline Sources/VLCPlaybackController.m]: Same~

File: Sources/VLCDocumentClass.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Authors: Carola Nitz <nitz.carola # googlemail.com>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import Foundation
+
+@objc(VLCDocumentClass)
+class DocumentClass: UIDocument {
```
Comment (Mikanbu) [inline Sources/VLCDocumentClass.swift]: Do we need this class or was it to prepare for future checks?

File: Sources/VLCAppDelegate.m [hunk 4]
```diff
@@ -353,7 +343,25 @@ - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDiction
...
+                        }
+                    } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+                        /* UIApplication's replacement calls require iOS 10 or later, which we can't enforce as of yet */
+                        if (success) {
+                            [[UIApplication sharedApplication] openURL:successCallback];
+                        } else {
+                            [[UIApplication sharedApplication] openURL:errorCallback];
+                        }
```
Comment (Mikanbu) [inline Sources/VLCAppDelegate.m]: How would you feel about simplifying the success checks by caching the callback?
Something like:

```
NSURL *callback = success ? successCallback : errorCallback;
if (@available(iOS 10, *)) {
    [[UIApplication sharedApplication] openURL:callback options:@{} completionHandler:nil];
} else {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    /* UIApplication's replacement calls require iOS 10 or later, which we can't enforce as of yet */
    [[UIApplication sharedApplication] openURL:callback];
#pragma clang diagnostic pop
```

File: Sources/VLCPlaybackController.h [hunk 5]
```diff
@@ -13,7 +13,8 @@
 
 #import "VLCEqualizerView.h"
 
-extern NSString *const VLCPlaybackControllerPlaybackDidStart;
+NS_ASSUME_NONNULL_BEGIN
```
Comment (Mikanbu) [inline Sources/VLCPlaybackController.h]: Shouldn't the nonnull be another commit which would come before this one? I feel that it's kind of wrong that this commit touches `VLCRendererDiscovererManager`.

File: Sources/VLCDocumentClass.swift (hunk 3 above)
Comment (carolanitz) [reply Sources/VLCDocumentClass.swift]: we need to have a subclass as mentioned in the Documentation. You can't use it directly

File: Sources/VLCAppDelegate.m (hunk 4 above)
Comment (carolanitz) [reply Sources/VLCAppDelegate.m]: yes definitely!

File: Sources/VLCDocumentClass.swift (hunk 3 above)
Comment (Mikanbu) [reply Sources/VLCDocumentClass.swift]: I see, thanks for the explanation!
